autoload/peekaboo.vim

```diff
@@ -64,7 +64,7 @@ endfunction
 " Appends macro list for the specified group to Peekaboo window
 function! s:append_group(title, regs)
   let compact = get(g:, 'peekaboo_compact', s:default_compact)
-  if !compact | call append(line('$'), a:title.':') | endif
+  if !compact && !empty(a:regs) | call append(line('$'), a:title.':') | endif
```

Comment:
Seems like we're still calling the last `append`. So I would instead just return early at the beginning of the function.